test/CMakeLists.txt

```diff
@@ -31,24 +35,26 @@ set( CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib )
 # Include filepaths for source and include.
 include( ${MODULE_ROOT_DIR}/sigv4FilePaths.cmake )
 
-# Target for Coverity analysis that builds the library.
-add_library( coverity_analysis
-            ${SIGV4_SOURCES} )
+if( COV_ANALYSIS )
```

Comment:
Do we need a coverity specific target at all? Coverity should be able to work with the normal targets.

Reply:
This is also an update to align FreeRTOS libraries design. It is true that normal targets can work with coverity. However, normal project may not configured properly to compile all the code we want to run coverity with. Another advantage of the coverity project is simplicity and independence.